maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/JarMojo.java

```diff
@@ -173,6 +222,79 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         project.getArtifact().setFile(jarFile);
     }
 
+    private void validateArchetypeScripts() throws MojoExecutionException {
+        File schemaFile = getArchetypeSchemaFile();
+        if (schemaFile == null) {
+            return;
+        }
+        try {
+            Validator validator = getValidator(schemaFile);
+            List<Path> xmlFiles = getXmlFiles();
+            DocumentBuilder db = getDocumentBuilder();
+            for (Path xmlFilePath : xmlFiles) {
+                File xmlFile = xmlFilePath.toFile();
+                Document doc = db.parse(xmlFile);
+                if (doc.getDocumentElement().getNodeName().equals(ARCHETYPE_ROOT_ELEMENT)) {
+                    validator.validate(new StreamSource(xmlFile));
+                }
+            }
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private DocumentBuilder getDocumentBuilder() throws ParserConfigurationException {
```

Comment:
Both `getDocumentBuilder` and `getValidator` are 3 line methods that are called only once and don't do anything special, please inline them inside `validateArchetypeScripts`.